src/main/java/org/folio/auth/authtokenmodule/tokens/TokenValidationException.java

```diff
@@ -0,0 +1,36 @@
+package org.folio.auth.authtokenmodule.tokens;
+
+/**
+ * When a token is provided and it hasn't passed validation, for any reason,
+ * a TokenValidationException will be present. Token validation exceptions have
+ * an opinion about what http response code should be returned to clients.
+ */
+public class TokenValidationException extends Exception {
+  /**
+   * Returns the HTTP response code to return when this exception is encountered.
+   * @return The HTTP response code.
+   */
+  public int getHttpResponseCode() {
+    return httpResponseCode;
+  }
+  private int httpResponseCode;
+
+  public TokenValidationException(String message, int httpResponseCode) {
+    super(message);
+    this.httpResponseCode = httpResponseCode;
+  }
+
+  public TokenValidationException(String message, Exception cause, int responseCode) {
+    super(message, cause);
+    httpResponseCode = responseCode;
+  }
+
+  @Override
+  public String toString() {
+    if (getCause() != null)
+      return String.format("TokenValidationException: %s %s", getMessage(), getCause().getClass());
```

Comment:
For readability curly braces around a single Java statement are never omitted. https://dev.folio.org/guidelines/contributing/#coding-style , https://www.oracle.com/java/technologies/javase/codeconventions-statements.html#15395

Reply:
Done